tests/application_development/code_relocation/test_lib/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 NXP
+
+add_library(test_lib STATIC "")
```

Comment:
I'm not particularly happy with how this library is declared- the goal is to add a library that has the same properties as an in tree Zephyr library, without needing to add a "dummy" library somewhere outside of the sample. I could also use the same approach Zephyr uses for external drivers to add these files, but felt that was unwieldy for a simple test.